azavea/raster-foundry
# Filter scenes by map bounds

## Overview

This filters the scenes in the scene list by the map boundaries. As the map is moved or zoomed, the boundaries will update and the scene list filtering will be updated. This now allows someone to zoom in to a particular area of interest and receive only scenes which cover that area. This also updates the Scenes API endpoint to accept coordinates in lat/lon for bounding boxes and points.

### Checklist


### Demo

![zoom1](https://cloud.githubusercontent.com/assets/447977/20015163/94e16f9a-a290-11e6-87ee-4d552f23df3c.png)

![zoom2](https://cloud.githubusercontent.com/assets/447977/20015169/986972d4-a290-11e6-844d-ae28734b55bd.png)

![zoom3](https://cloud.githubusercontent.com/assets/447977/20015172/9d1d2ed8-a290-11e6-9795-776697cffffd.png)

### Notes

I did some refactoring to bring things more in line with my understanding of Angular best practices with an eye toward making an eventual migration to Angular 2 easier, but I'm lacking in project context, so if I changed something that had a good reason for being the way it was, let me know. It would also be good to click around and make sure I haven't broken anything unrelated; I think everything works the way it did before but I might have missed something.

Also note that I discovered that the `index.routes.js` file was using DOS line endings so I converted it to Unix; I've broken that commit out separately so it may be helpful to view the commits individually to get a better sense of what actually changed.

## Testing Instructions

 * From inside the vagrant machine, `./scripts/server`
 * `cd app-frontend && npm start`
 * Navigate to http://localhost:9100/#/browse/
 * Zoom and pan the map and note that the filtered scenes update as the map view changes.

Closes #618
